src/blueprint/from_config/operators/expr.rs

```diff
@@ -224,5 +224,607 @@ mod tests {
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_logic_cond() {
+    let expected = json!(0);
+
+    let actual = Expr::eval(
+      json!({"body": {"cond": [{"const": 0}, [[{"const": false}, {"const": 1}], [{"const": false}, {"const": 2}]]]}}),
+    )
+    .await
+    .unwrap();
+    assert_eq!(actual, expected);
+
+    let expected = json!(1);
+
+    let actual = Expr::eval(
+      json!({"body": {"cond": [{"const": 0}, [[{"const": true}, {"const": 1}], [{"const": true}, {"const": 2}]]]}}),
+    )
+    .await
+    .unwrap();
+    assert_eq!(actual, expected);
+
+    let expected = json!(2);
+    let actual = Expr::eval(
+      json!({"body": {"cond": [{"const": 0}, [[{"const": false}, {"const": 1}], [{"const": true}, {"const": 2}]]]}}),
+    )
+    .await
+    .unwrap();
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_logic_default_to() {
+    let expected = json!(0);
+    let actual = Expr::eval(json!({"body": {"defaultTo": [{"const": null}, {"const": 0}]}}))
+      .await
+      .unwrap();
+    assert_eq!(actual, expected);
+
+    let expected = json!(true);
+    let actual = Expr::eval(json!({"body": {"defaultTo": [{"const": ""}, {"const": true}]}}))
+      .await
+      .unwrap();
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_concat() {
+    let expected = json!([1, 2, 3, 4]);
+    let actual = Expr::eval(json!({"body": {"concat": [{"const": [1, 2]}, {"const": [3, 4]}]}}))
+      .await
+      .unwrap();
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_relation_intersection() {
+    let expected = json!([3]);
+    let actual = Expr::eval(json!({"body": {"intersection": [{"const": [1, 2, 3]}, {"const": [3, 4, 5]}]}}))
+      .await
+      .unwrap();
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_relation_difference() {
+    let expected = json!([1]);
+    let actual = Expr::eval(
+      json!({"body": {"difference": [[{"const": 1}, {"const": 2}, {"const": 3}], [{"const": 2}, {"const": 3}]]}}),
+    )
+    .await
+    .unwrap();
+
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_relation_simmetric_difference() {
+    let expected = json!([1]);
+
+    let actual = Expr::eval(
+      json!({"body": {"symmetricDifference": [[{"const": 1}, {"const": 2}, {"const": 3}], [{"const": 2}, {"const": 3}]]}}),
+    )
+    .await
+    .unwrap();
+    assert_eq!(actual, expected);
+
+    let actual = Expr::eval(
+      json!({"body": {"symmetricDifference": [[{"const": 2}, {"const": 3}], [{"const": 1}, {"const": 2}, {"const": 3}]]}}),
+    )
+    .await
+    .unwrap();
+    assert_eq!(actual, expected);
+  }
+
+  #[tokio::test]
+  async fn test_relation_union() {
+    let expected = json!([1, 2, 3]);
+    let actual = Expr::eval(json!({"body": {"union": [[{"const": 1}, {"const": 2}], [{"const": 2}, {"const": 3}]]}}))
+      .await
+      .unwrap();
+    assert_eq!(actual, expected);
```

Comment:
we didn't had tests for these previously (on `test-expr-relation.yml`), after adding it to this file, the test seems flaky.

sometimes the value is `json!([1, 2, 3])`, and sometimes is `json!([2, 1, 3])`.

is there a way we can make the expression evaluate to a consistent value? @shashitnak @tusharmath